src/cmd/serve.rs

```diff
@@ -690,52 +704,85 @@ pub fn serve(
 
     loop {
         match rx.recv() {
-            Ok(event) => {
-                let can_do_fast_reload = !matches!(event, Remove(_));
+            Ok(events) => {
+                let mut meaningful_events: Vec<(PathBuf, SimpleFSEventKind, Instant)> = vec![];
+
+                // Bin all changes we care about over the last debounce period.
+                for event in events.as_ref().unwrap().iter() {
+                    let simple_kind = get_relevant_event_kind(&event.event.kind);
+                    if simple_kind.is_none() {
+                        continue;
+                    }
 
-                match event {
-                    // Intellij does weird things on edit, chmod is there to count those changes
-                    // https://github.com/passcod/notify/issues/150#issuecomment-494912080
-                    Rename(_, path) | Create(path) | Write(path) | Remove(path) | Chmod(path) => {
-                        if is_ignored_file(&site.config.ignored_content_globset, &path) {
-                            continue;
-                        }
+                    // We currently only handle notify events that report a single path per event.
+                    // Enforce that assumption to avoid surprises. See `get_relevant_event_kind`.
+                    assert!(
+                        event.event.paths.len() == 1,
+                        "Didn't expect notify event with multiple paths."
+                    );
```

Comment:
Either `get_relevant_event_kind` should make sure of that or we should return an error rather than a panic